### ISSUE TITLE
Sort-comp rule doesn't enforce getDerivedStateFromProps placement

### ISSUE THREAD

**Issue by lynxtaa** [yannickcr/eslint-plugin-react#1793]
With "react/sort-comp": "warn" provided code shouldn't be considered valid:
```jsx
static getDerivedStateFromProps(nextProps, prevState) {}

state = {}
```

Tested on version 7.8.2

**Comment by ljharb** [comment]
What is your full eslint config?

**Comment by ljharb** [comment]
@lynxtaa you should be specifying a version pragma in "settings" (see the read me), but I don't think that's it. Could you also share the entire component code in question?

**Comment by lynxtaa** [comment]
Still the same issue with specified pragma.

Example component `Test.jsx`:
```jsx
import React from 'react'

class Test extends React.Component {
	static getDerivedStateFromProps() {
		return null
	}

	constructor(props) {
		super(props)
		this.state = { test: 'test' }
	}

	render() {
		return this.state.test
	}
}

export default Test
```

React 16.3.2.

**Comment by ljharb** [comment]
ok, what warning do you get with that component?

**Comment by lynxtaa** [comment]
No warning. But i should get "react/sort-comp" warning according to this https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/sort-comp.md 

getDerivedStateFromProps should be declared after constructor.

**Comment by ljharb** [comment]
aha, this seems to be a bug in the sort-comp implementation then - since the lifecycle methods are typically instance methods, but `getDerivedStateFromProps` is a static method.

cc @joe-denea / #1771 which has tests for the "valid" case for this method, but not for the "invalid" case.

**Comment by joe-denea** [comment]
Sorry about that, I should definitely have tested that.

**Comment by ljharb** [comment]
Fixed in #1795.
